src/main/java/edu/harvard/iq/dataverse/DvObject.java

```diff
@@ -467,4 +484,14 @@ public void setStorageIdentifier(String storageIdentifier) {
     
     @OneToMany(mappedBy = "definitionPoint",cascade={ CascadeType.REMOVE, CascadeType.MERGE,CascadeType.PERSIST}, orphanRemoval=true)
     List<RoleAssignment> roleAssignments;
+    
+    /**
+     * Should only be used in constructors for DvObjectContainers (Datasets and 
+     * Collections), to make sure new entries are created and persisted in the 
+     * database StorageUse table for every DvObject container we create.
+     * @param storageUse 
+     */
+    public void setStorageUse(StorageUse storageUse) {
```

Comment:
if only possible for Dataverse and Dataset, is it possible to move this to DvObjectContainer? Or is that not possible for the db table to be properly created / connected to? Partially asking because I see the imports in DvObjectContaine, but no other references, so maybe you tried it there, but had to move it here? (if so, we should clean up those imports).

Reply:
To be clear, there was actually a reason I ended up putting it in DvObject - because the column in StorageUse needs an entity to point to, and DvObjectContainer is not an entity... But it was still possible to move the above to DvObjectContainer, I just had to keep this in StorageUse: 
```
@OneToOne
@JoinColumn(nullable=false)
private DvObject dvObjectContainer; 
```